Libraries/LibWeb/MimeSniff/Resource.cpp

```diff
@@ -183,6 +183,138 @@ bool matches_mp4_signature(ReadonlyBytes sequence)
     return false;
 }
 
+struct VintResult {
+    u64 value;
+    size_t length;
+};
+
+// https://mimesniff.spec.whatwg.org/#parse-a-vint
+VintResult parse_a_vint(ReadonlyBytes sequence, size_t iter)
+{
+    // To parse a vint on a byte sequence sequence of size length, starting at index iter use the following steps:
+    sequence = sequence.slice(iter);
+    auto length = sequence.size();
+
+    // 1. let mask be 128.
+    u8 mask = 128;
+
+    // 2. Let max vint length be 8.
+    size_t const max_vint_length = 8;
+
+    // 3. Let number size be 1.
+    size_t number_size = 1;
+
+    // 5. Let index be 0.
+    // NOTE: spec declares this after step four, but it is needed before to use in the loop.
+    size_t index = 0;
+
+    // 4. While number size is less than max vint length, and less than length, continuously loop through these steps:
+    while (number_size < max_vint_length && number_size < length) {
+        // 1. If the sequence[index] & mask is not zero, abort these steps.
+        if (sequence[index] & mask)
+            break;
+
+        // 2. Let mask be the value of mask >> 1.
+        mask >>= 1;
+
+        // 3. Increment number size by one.
+        number_size++;
+    }
+
+    // 6. Let parsed number be sequence[index] & ~mask.
+    u64 parsed_number = sequence[index] & ~mask;
+
+    // 7. Increment index by one.
+    index++;
+
+    // 8. Let bytes remaining be the value of number size - 1.
+    auto bytes_remaining = number_size - 1;
+
+    // 9. While bytes remaining is not zero, execute there steps:
+    while (bytes_remaining != 0) {
+        // 1. Let parsed number be parsed number << 8.
+        parsed_number <<= 8;
+
+        // 2. Let parsed number be parsed number | sequence[index].
+        parsed_number |= sequence[index];
+
+        // 3. Increment index by one.
+        index++;
+
+        // 4. If index is greater or equal than length, abort these steps.
+        if (index >= length)
+            break;
+
+        // 5. Decrement bytes remaining by one.
+        bytes_remaining--;
+    }
+
+    // 10. Return parsed number and number size
+    return { parsed_number, number_size };
+}
+
+// https://mimesniff.spec.whatwg.org/#signature-for-webm
+bool matches_webm_signature(ReadonlyBytes sequence)
+{
+    // 1. Let sequence be the byte sequence to be matched, where sequence[s] is byte s in sequence and sequence[0] is the first byte in sequence.
+
+    // 2. Let length be the number of bytes in sequence.
+    auto length = sequence.size();
+
+    // 3. If length is less than 4, return false.
+    if (length < 4)
+        return false;
+
+    // 4. If the four bytes from sequence[0] to sequence[3], are not equal to 0x1A 0x45 0xDF 0xA3, return false.
+    static auto const webm_signature = "\x1A\x45\xDF\xA3"sv.bytes();
```

Comment:
if possible this could be `static auto constexpr` rather than just const. super nitpick though.